spec/controllers/books_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+require './spec/shared_context.rb'
+
+RSpec.describe "Books", type: :controller do
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 5) }
+
+      it 'responses with the books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'When fetching a book' do
+      let!(:book) { create(:book) }
+
+      it 'responses with the book json' do
```

Comment:
Here I think you should add the param id to the test, because this ep returns a specific book with the param no?.